src/com/duckduckgo/mobile/android/activity/OnboardingActivity.java

```diff
@@ -115,12 +109,7 @@ public void onPageScrollStateChanged(int state) {
         });
         viewPager.setAdapter(adapter);
 
-        int[] backgroundColors = new int[] {
-                ContextCompat.getColor(this, PrivacyFragment.BACKGROUND_COLOR),
-                ContextCompat.getColor(this, NoAdsFragment.BACKGROUND_COLOR),
-                ContextCompat.getColor(this, NoTrackingFragment.BACKGROUND_COLOR),
-                ContextCompat.getColor(this, RightFragment.BACKGROUND_COLOR),
-                ContextCompat.getColor(this, EndOnboardingFragment.BACKGROUND_COLOR)};
+        int[] backgroundColors = Onboarding.getBackgroundColors(this);
```

Comment:
I like this change. You can take this a step further and have the adapter provide the correct color resource (which the activity can convert using ContextCompat) for a given position. The adapter is already responsible for generating the OboardingValues which already include the color. If these are stored in an array in the adapter (as per other suggestions) you can request the color from there rather than having the color info duplicated.